### PR TITLE
Cryptohash is deprecated. Use Cryptonite instead.

### PR DESCRIPTION
**Pull request by danielrutz**
P.S. Probably my first real line of Haskell. Don't be that harsh to me if it is bad style.

### CODE REVIEW

**Comment by JohnAZoidberg** [comment]
Wow nice, thanks. Great spot!

File: app/Util.hs [hunk 1]
```diff
@@ -42,7 +43,7 @@ decodeHex = fst . B16.decode . E.encodeUtf8
 
 hashPassword :: T.Text -> BS.ByteString -> T.Text
 hashPassword password salt =
-     makeHex . SHA.finalize $ SHA.updates SHA.init [salt, E.encodeUtf8 password]
+     makeHex $ BA.convert . Hash.hashFinalize $ Hash.hashUpdates (Hash.hashInitWith Hash.SHA512) [salt, E.encodeUtf8 password]
```
Comment (JohnAZoidberg) [inline app/Util.hs]: The first function application ($) can be written as function composition (.).
I think while we're at it, we should switch to a KDF like Scrypt, which is implemented in Cryptonite as well.

File: Flatr-App-Core.cabal [hunk 2]
```diff
@@ -36,7 +36,8 @@ executable Flatr-App-Core-exe
                      , word8
                      , bytestring
                      , base16-bytestring >=0.1
-                     , cryptohash >=0.11
+                     , cryptonite >=0.24
+                     , memory
```
Comment (JohnAZoidberg) [inline Flatr-App-Core.cabal]: What is memory required for?

File: Flatr-App-Core.cabal [hunk 3]
```diff
@@ -36,7 +36,8 @@ executable Flatr-App-Core-exe
                      , word8
                      , bytestring
                      , base16-bytestring >=0.1
-                     , cryptohash >=0.11
+                     , cryptonite >=0.24
```
Comment (JohnAZoidberg) [inline Flatr-App-Core.cabal:39]: Please add this dependency to the nix file.

**Review by JohnAZoidberg — CHANGES_REQUESTED** [review]
Great first contribution!
Please address my comments. :)

File: Flatr-App-Core.cabal (hunk 2 above)
Comment (danielrutz) [reply Flatr-App-Core.cabal]: For Data.ByteArray.convert, which is used to convert the Context given by hashFinalize to a ByteString.

File: Flatr-App-Core.cabal (hunk 3 above)
Comment (danielrutz) [reply Flatr-App-Core.cabal:39]: Fixed in Commit e4efd16. Not tested, however.

File: app/Util.hs (hunk 1 above)
Comment (danielrutz) [reply app/Util.hs]: Initial implementation to be found in 6d2931b. Keep in mind, however, that this breaks compatibility with earlier commits.

File: Flatr-App-Core.cabal (hunk 2 above)
Comment (danielrutz) [reply Flatr-App-Core.cabal]: No longer needed, deleted from the cabal file.

**Pull request merged**